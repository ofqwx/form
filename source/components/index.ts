export { FormContext } from './Form/FormProvider';
export { default as FormProvider } from './Form/FormProvider';
export { default as Field } from './Field/Field';
export { default as Form } from './Form/Form';
