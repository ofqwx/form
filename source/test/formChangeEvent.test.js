/**
 * @jest-environment jsdom
 */

import React from 'react';
import { Form, Input } from '../components';
import { render, fireEvent, screen } from '@testing-library/react';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import '@testing-library/jest-dom/extend-expect';

expect.extend({ toMatchDiffSnapshot });

test('Value change when change event is triggered', async () => {
  const { asFragment } = render(
    <Form initialValues={{ firstName: 'Antonio' }} onSubmit={() => undefined}>
      <Input label="First name" name="firstName" />

      <button type="submit">Submit</button>
    </Form>
  );

  const firstRender = asFragment();

  fireEvent.change(screen.getByRole('textbox'), {
    target: { value: 'Pedro' },
  });

  expect(firstRender).toMatchDiffSnapshot(asFragment());
});
