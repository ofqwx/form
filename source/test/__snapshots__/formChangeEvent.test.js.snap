// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value change when change event is triggered 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -5,11 +5,11 @@
      </label>
      <br />
      <input
        name=\\"firstName\\"
        type=\\"text\\"
-       value=\\"Antonio\\"
+       value=\\"Pedro\\"
      />
      <button
        type=\\"submit\\"
      >
        Submit"
`;
